### PR TITLE
test support go1.18

### PR DESCRIPTION
**Pull request by thinkerou**
await go1.18 released.

ref:
1. https://go.dev/blog/go1.18beta1
2. https://github.com/golang/go/commit/2580d0e08d5e9f979b943758d3c49877fb2324cb

### CODE REVIEW

**Comment by jincheng9** [comment]
nice try~

**Comment by thinkerou** [comment]
@appleboy please help me review, thanks!

File: any.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2021 Gin Core Team.  All rights reserved.
```
Comment (appleboy) [inline any.go:1]: 2021 -> 2022

File: binding/any.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2021 Gin Core Team.  All rights reserved.
```
Comment (appleboy) [inline binding/any.go:1]: same as above

**Review by appleboy — APPROVED** [review]

File: any.go [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2022 Gin Core Team.  All rights reserved.
+// Use of this source code is governed by a MIT style
+// license that can be found in the LICENSE file.
+
+//go:build !go1.18
+// +build !go1.18
```
Comment (shaba) [inline any.go:6]: May be invert?
//go:build go1.18
// +build go1.18

binding.go:33:22: undeclared name: any (requires version go1.18 or later)

Comment (alvian-pratama) [reply any.go:6]: +1, 
this caused the above issue if the machine has go version 1.18 but the app that imports gin uses version lower than 1.18

**Pull request merged**